pkg/server/proto/server.proto

```diff
@@ -1087,6 +1105,32 @@ message TaskLaunchInfo {
   repeated string arguments = 3;
 }
 
+// TrackTask tracks the life of an on-demand runner task that spawns Start
+// and Stop tasks for any kind of job/operation in Waypoint. Automatic jobs
+// such as project polling or status report generation spawn on-demand
+// runner tasks, and this message can be used to track the life of those
+// automated jobs. Note that every on-demand runner task is wrapped with a
+// Start and Stop task, which we track here as well.
+message TrackTask {
+  // The ID of this message. If on first upsert, the id does not need to be
+  // specified and the state pkg will autogenerate an id. Specifying an id
+  // assumes the TrackTask message already exists in the database.
+  string id = 1;
+
+  // The referred job stores data like ODR config, the target runner used,
+  // timestamp for when job started or completed, operation state for task, etc.
+  // Currently the only required field when Upserting a TrackTask.
+  Ref.Job task_job = 2;
+
+  // Both start_job and stop_job relate to task_job as these jobs were
+  // responsibile for starting and stopping the task job.
+  Ref.Job start_job = 3;
+  Ref.Job stop_job = 4;
+
+  // This task info message represented as json format
+  string task_json = 5;
```

Comment:
What is this?

Reply:
This was the metadata json suggestion for the frontend to consume raw task info like `artifact_json`. Not hooked up to anything yet.

Reply:
Done!